sdk/core/core-http/src/policies/userAgentPolicy.ts

```diff
@@ -59,9 +59,11 @@ export function getDefaultUserAgentValue(): string {
 
 export function userAgentPolicy(userAgentData?: TelemetryInfo): RequestPolicyFactory {
   const key: string =
-    !userAgentData || userAgentData.key == undefined ? getDefaultUserAgentKey() : userAgentData.key;
+    !userAgentData || userAgentData.key === undefined
+      ? getDefaultUserAgentKey()
+      : userAgentData.key;
   const value: string =
```

Comment:
could do 

```ts
const value = userAgentData?.value ?? getDefaultUserAgentKey()
```

instead, much more elegant

Reply:
I'm not sure why, but this doesn't seem to be working. I believe improvements like this should be done separately. This PR is about ESLint fixes, and it's getting hard to make it work even with that limited scope.

Reply:
Sure, I was feeling adventurous since I hadn't seen a spot that `??` could be useful before.